tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/util/NginxUtil.java

```diff
@@ -27,58 +32,95 @@
+
         // we have to copy the local paths from the host machine into the K3s container,
         // since this is the source used for the hostPath volumes
         for (final var localPath : localPaths) {
             container.copyFileToContainer(MountableFile.forHostPath(localPath), localPath.toAbsolutePath().toString());
         }
 
-        if (withBasicAuth) {
+        if (withHttps) {
+            final var nginxSecret = createNginxCertificateSecret(namespace);
+            client.resource(nginxSecret).create();
+
+            container.copyFileToContainer(MountableFile.forClasspathResource("nginx/nginx-https.conf"),
+                    "/nginx-https-conf");
+        } else if (withBasicAuth) {
             container.copyFileToContainer(MountableFile.forClasspathResource("nginx/nginx-htpasswd"),
                     "/nginx-htpasswd");
-            container.copyFileToContainer(MountableFile.forClasspathResource("nginx/nginx-default.conf"),
-                    "/nginx-default-conf");
+            container.copyFileToContainer(MountableFile.forClasspathResource("nginx/nginx-basic-auth.conf"),
+                    "/nginx-basic-auth-conf");
         }
 
-        final var nginxDeployment = createNginxDeployment(namespace, localPaths, withBasicAuth);
+        final var nginxDeployment = createNginxDeployment(namespace, localPaths, withHttps, withBasicAuth);
         client.resource(nginxDeployment).create();
 
         final var nginxService = createNginxService(namespace);
         client.resource(nginxService).create();
 
         await().untilAsserted(() -> {
-            final var nginx = client.apps().deployments().inNamespace(namespace).withName(NGINX_APP_NAME).get();
-            assertThat(nginx).as("%s deployment", NGINX_APP_NAME).isNotNull();
-            assertThat(nginx.getStatus()).as("%s status", NGINX_APP_NAME).isNotNull();
+            final var nginx = client.apps().deployments().inNamespace(namespace).withName(NGINX_CONTAINER_NAME).get();
+            assertThat(nginx).as("%s deployment", NGINX_CONTAINER_NAME).isNotNull();
+            assertThat(nginx.getStatus()).as("%s status", NGINX_CONTAINER_NAME).isNotNull();
             assertThat(nginx.getStatus().getReadyReplicas()).isEqualTo(1);
         });
     }
 
+    private static @NotNull Secret createNginxCertificateSecret(final @NotNull String namespace) {
+        return new SecretBuilder().withType("kubernetes.io/tls")
+                .withNewMetadata()
+                .withNamespace(namespace)
+                .withName("nginx-certificates")
+                .endMetadata()
+                .addToData("tls.crt", getResourceAsBase64("/nginx/nginx-tls.crt"))
+                .addToData("tls.key", getResourceAsBase64("/nginx/nginx-tls.key"))
+                .build();
+    }
+
+    private static String getResourceAsBase64(final @NotNull String name) {
+        try {
+            final var resourcePath = Path.of(NginxUtil.class.getResource(name).toURI());
```

Comment:
Just to avoid the warning on IntelliJ regarding the potential NPE in the `toURI()` invocation, we could add the annotation `@SuppressWarnings("DataFlowIssue")` in this method.

Reply:
Will replace it with a pulled out variable + not-null assertion.